.github/workflows/publish.yaml

```diff
@@ -6,17 +6,15 @@
 # separate terms of service, privacy policy, and support
 # documentation.
 
-name: Upload Python Package
+name: Build & Optional Deploy
 
-on:
-  release:
```

Comment:
We won't produce github releases, and we want to check if package building works during pull requests.

Also this is why this job was ignored during previous pull-requests and previous push.
Now you can observe that this job is triggered and you can observe that the deployment steps are ignored on pull requests.